script.js

```diff
@@ -1,4 +1,207 @@
+  "GAME_MODE_CHOOSE_DICE_ORDER_AUTOMATICALLY";
+
+var numDiceChosen = 0;
+
+var numPlayers = 0;
+
+var gameMode = GAME_MODE_CHOOSE_NUM_PLAYERS;
+
+var currPlayer = 0;
+
+var playerProfiles = [];
+
+var roundWinner = null;
+
+var getDiceRoll = function () {
+  return Math.ceil(Math.random() * 6);
+};
+
+var getDiceRolls = function () {
+  var newDiceRolls = [];
+  for (var i = 0; i < numDiceChosen; i += 1) {
+    newDiceRolls.push(getDiceRoll());
+  }
+
+  playerProfiles[currPlayer].diceRolls = newDiceRolls;
+
+  return newDiceRolls;
+};
+
+var concatenate2Numbers = function (num1, num2) {
+  return Number(String(num1) + String(num2));
+};
+
+var getPlayerNumberAutomatically = function () {
+  var diceArray = playerProfiles[currPlayer].diceRolls;
+  var playerNum;
+
+  diceArray.sort(function (a, b) {
+    return a - b;
+  });
+
+  while (diceArray.length > 0) {
+    var largestNum = diceArray.pop();
+    if (playerNum == null) {
+      playerNum = largestNum;
+    } else {
+      playerNum = concatenate2Numbers(playerNum, largestNum);
+    }
+  }
+  playerProfiles[currPlayer].diceNum = playerNum;
+  return playerNum;
+};
+
+var determineWinner = function () {
+  if (roundWinner == null) {
+    roundWinner = playerProfiles[currPlayer];
+  } else if (playerProfiles[currPlayer].diceNum > roundWinner.diceNum) {
+    roundWinner = playerProfiles[currPlayer];
+  }
+};
+
+var addNumToRunningScore = function (roundScore) {
+  playerProfiles[currPlayer].score += roundScore;
+};
+
+var createLeaderBoardOutput = function () {
+  var leaderBoardOutput = "Leaderboard: ";
+
+  playerProfiles.sort((a, b) => b.score - a.score);
+  for (var i = 0; i < playerProfiles.length; i += 1) {
+    leaderBoardOutput +=
+      "<br> Player " + playerProfiles[i].id + ": " + playerProfiles[i].score;
+  }
+  return leaderBoardOutput;
+};
+
+var resetGame = function () {
+  currPlayer = 0;
+  gameMode = GAME_MODE_CHOOSE_NUM_DICE;
+  roundWinner = null;
+};
+
+var createDiceRollInfoMsg = function (diceRollsArr) {
+  var outputMsg =
+    "Welcome Player " +
+    playerProfiles[currPlayer].id +
+    ".<br> Your dice rolls are:";
+
+  for (var i = 0; i < diceRollsArr.length; i += 1) {
+    outputMsg += "<br> Dice " + (i + 1) + ": " + diceRollsArr[i];
+  }
+  return outputMsg;
+};
+
+var createPlayerProfiles = function () {
+  for (var i = 0; i < numPlayers; i += 1) {
+    playerProfiles.push({ id: i + 1, diceRolls: [], diceNum: 0, score: 0 });
+  }
+};
+var totalReset = function () {
+  currPlayer = 0;
+  gameMode = GAME_MODE_CHOOSE_NUM_PLAYERS;
+  roundWinner = null;
+  playerProfiles = [];
+  numPlayers = 0;
+  numDiceChosen = 0;
+};
+//
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log(gameMode);
+  console.log(currPlayer);
+  if (gameMode == GAME_MODE_CHOOSE_NUM_PLAYERS) {
+    if (isNaN(input) == true) {
+      return "Please enter a number between 2 players to 5 players";
+    }
+    if (Number(input) < 2) {
```

Comment:
You seem to be converting input to Number very often and then comparing it to another Number. You can consider creating a function for this. e.g.
```js
function isLessThan(input, num) {
// logic here
}
```